### ISSUE TITLE
Hi there - attempting to integrate with Emby (dotnet 3.1.4)

### ISSUE THREAD

**Issue by AnthonyMusgrove** [jstedfast/MailKit#1031]
**Describe the bug**
I've compiled both MailKit and MimeKit from source, copied both of the .dlls to my project (plugin) directory, referenced to them, and when I deploy the plugin I continue to receive this:

System.MissingMethodException: Method not found: 'Void MailKit.IMailTransport.Send(MimeKit.MimeMessage, System.Threading.CancellationToken, MailKit.ITransferProgress).

No matter what I do, it continues to state this, even though Emby does not use Mailkit at all, so I'd be introducing it.  I'm trying to update this project: https://github.com/MediaBrowser/email-notifications to use MailKit rather than SmtpClient (because its incompatible with a lot of the user's email providers).

Unfortunately I can't just install from nuget, because I have to embed the compiled dlls into the plugin dll and use reflection to introduce them.  It works fine on my windows computer locally if I create a new windows forms project, and sends emails successfully.  But this plugin/dotnet core 3.1.4 is doing my head in. :(

**Platform (please complete the following information):**
 - OS: Windows
 - .NET Runtime: 3.1.4
 - .NET Framework: 
 - MailKit Version: Latest.  

Any advice would be so greatly appreciated.  I'm stuck now trying to get this integrated ! :/

**Comment by jstedfast** [comment]
I don't know, but I would guess that the problem is some sort of code-wise mixup of System.Net.Mail.SmtpClient and MailKit.Net.Smtp.SmtpClient. Make sure to remove all references to System.Net.Mail in the code.

**Comment by AnthonyMusgrove** [comment]
Thank you mate, this gives me a starting point.

**Comment by AnthonyMusgrove** [comment]
It looks like emby uses dotnetcore clr and ive been trying to include the standard 2.0 dlls.  I take it this would be my issue

**Comment by jstedfast** [comment]
you mean the `netstandard2.0` dll's? If so, those are the correct ones. `netstandard2.0` is the name for the dotnet core library profile. Then you have `netcoreapp3.1`, for example, which is a profile for dotnet core apps.

**Comment by AnthonyMusgrove** [comment]
Yep those are the ones Ive included in the project.  Basically added dependencies for Mailkit.dll and Mimekit.dll and getting that odd issue.  Ill see if I can debug a little more this afternoon.  Im just very glad to know that's far Ive used the correct dlls and that dotnet core is supported.  Now I just have to try to connect the dots.  Thank you so much for taking the time to give me some advice :)

**Comment by AnthonyMusgrove** [comment]
I just thought I'd let you know that I resolved the issue.  It was my dependency injections.  I use AppDomain.AssemblyResolve, however I was reloading the assembly each time it was requested by this callback.  And I noticed that MailKit calls MimeKit to be loaded, then my software etc, and it stacks.  To resolve the issue, I simply load the assembly into an Assembly object (for MimeKit and MailKit), then when requested, simply return the loaded Assembly object rather than reloading for each call.   And vwalah; it works awesomely, thank you so much again for your help, it has been unbelievably useful.

https://github.com/AnthonyMusgrove/email-notifications
